layout/js/animate.js

```diff
@@ -36,17 +42,35 @@ var animatePrcp = function(timestep, $currentStormDot) {
       $currentStormDot.data('cx', stormX);
       $currentStormDot.data('cy', stormY);
     }
+
   });
 
-  $('.nwis-dot').css('fill', '#73877B').css('stroke', "#385F71");
-  $('.f-' + timestep).css('fill', '#ff7000');
-  $('#timestamp-text').html(prcpTimes.times[timestep - 1]);
+  $('.nwis-dot').css('fill', '#057083').css('stroke', "#057083");
+  $('.f-' + timestepLocal).css('fill', 'red');
+  $('#timestamp-text').html(prcpTimes.times[timestepLocal - 1]);
+
 
-  var darkWidth = (timestep+1)/prcpTimes.times.length;
+  var darkWidth = (timestepLocal+1) / prcpTimes.times.length;
   $('#spark-light-mask').attr('x', darkWidth).attr('width', 1 - darkWidth);
   $('#spark-full-mask').attr('width', darkWidth);
   $('#flood-light-mask').attr('x', darkWidth).attr('width', 1 - darkWidth);
   $('#flood-full-mask').attr('width', darkWidth);
+
+  // requestAnimationFrame BASE CASE
+  if (timestepLocal < prcpTimes.times.length) {      
+    if (running) {
+      window.requestAnimationFrame(function() {
+        animatePrcp(timestepLocal + 1);
```

Comment:
I think what you want is rather than incrementing timestepLocal in this way, run the animation frame as a separate piece that calculates the percentage through the animation and call `animatePrcp()` on the index that corresponds to that percentage.

Reply:
Ahh I see. This is a better approach! Then slower devices can skip timesteps if necessary and still be animating at the same pace as a faster device. Thanks for the help!

Reply:
wow, smart!